### PR TITLE
Fix 145. Registration with same phone email password value. narek

### PR DESCRIPTION
**Pull request by narekacy**
Before you start reviewing my pull request:
1. Sorry for napolitana spaghetti code in `register.js`.
2. Sorry for napolitana spaghetti code in `register.js`.

**In this pull request next fixes are done**:

— impossibility to register an account with 
- password with same value as phone number;
- impossibility to register an account with password with same value as email;
- impossibility to register an account with email or phone number that are already registered;

— unit tests for all fixes are added. 

Again, sorry for spaghetti code in `register.js`.

_King regards,
Narek_

**EDITED 25.04.20**
I removed tests as I'm not able to create them as it is needed yet. If anybody wants to complete them, I will study it with pleasure.

### CODE REVIEW

**Comment by vitaliimak** [comment]
Could you update current dev brunch and fix conflicts?

**Comment by narekacy** [comment]
> Could you update current dev brunch and fix conflicts?

Done

File: server/src/test/java/com/lunchtime/service/impl/PersonServiceImplTest.java [hunk 1]
```diff
@@ -31,6 +31,38 @@ public void setUp() {
...
+            .phoneNumber("+380501234567")
+            .password("+380501234567")
+            .build();
+
+        String phoneNumber = person.getPhoneNumber();
+        String password = person.getPassword();
+        if (phoneNumber.equals(password)) {
+            person = null;
+        }
+        Assert.assertNull(person);
```
Comment (vitaliimak) [inline server/src/test/java/com/lunchtime/service/impl/PersonServiceImplTest.java]: You should test your method in service, not local variable

You can check if your test works. Comment lines 26-29 in PersonServiceImpl.java, your tests will be green

File: client/src/styles/no-match.css [hunk 2]
```diff
@@ -9,3 +9,13 @@
 .m-button {
   font-size: 100%;
 }
+
+.m-button:hover{
```
Comment (vitaliimak) [inline client/src/styles/no-match.css:13]: Why don't you use standard button from theme?

Why you add this code in this PR? If you want to change this button style, you should create task and make independent PR

**Review by vitaliimak — CHANGES_REQUESTED** [review]
You should divide your PR to 2 separate PRs (button style and registration) 

And you should fix tests

**Comment by narekacy** [comment]
> You should divide your PR to 2 separate PRs (button style and registration)

I know. Next time)

> And you should fix tests

What's wrong with tests?

File: client/src/styles/no-match.css (hunk 2 above)
Comment (narekacy) [reply client/src/styles/no-match.css:13]: Yes. Next time. 
Thanks

**Comment by vitaliimak** [comment]
> What's wrong with tests?

Emm... https://github.com/Social-projects-Rivne/Lunch_Time/pull/75#discussion_r414026677
They don't work

File: server/src/test/java/com/lunchtime/controllers/RegisterPersonControllerTest.java [hunk 3]
```diff
@@ -84,4 +90,18 @@ public void create_not_null_person_expected_true() throws Exception {
...
+
+    @Test
+    public void return_null_if_phone_number_equals_password() throws Exception {
+        registerPerson.setPassword(registerPerson.getPhoneNumber());
+        ResponseEntity<?> person = personController.createPerson(registerPerson);
+        Assert.assertNull(person);
+    }
+
+    @Test
+    public void return_null_if_email_equals_password() throws Exception {
```
Comment (vitaliimak) [inline server/src/test/java/com/lunchtime/controllers/RegisterPersonControllerTest.java]: image
image

What this test do? It always green. Every developer can change creating person code, but your tests don't show error

File: server/src/main/java/com/lunchtime/controllers/PersonController.java [hunk 4]
```diff
@@ -20,33 +18,27 @@
...
-                .build();
+            return null;
         }
 
         PersonDto personDto = null;
         try {
             personDto = personService.saveRegisterPerson(registerPerson);
         } catch (NonUniqueResultException nue) {
-            return ResponseEntity.status(Integer.parseInt(nue.getMessage())).build();
+            return ResponseEntity.status(409)
```
Comment (vitaliimak) [inline server/src/main/java/com/lunchtime/controllers/PersonController.java:34]: https://httpstatuses.com/409

File: server/src/main/java/com/lunchtime/controllers/PersonController.java [hunk 5]
```diff
@@ -20,33 +18,27 @@
...
         }
-        
+
         if (personDto != null) {
-            JwtPersonDetails jwtPersonDetails = new JwtPersonDetails(
-                registerPerson.getEmail(), registerPerson.getPassword());
-            authController.createAuthenticationToken(jwtPersonDetails);
-            String token = tokenHistory.getTokenList().remove(0);
-            return ResponseEntity.ok(token);
+            return ResponseEntity.ok(personDto);
```
Comment (vitaliimak) [inline server/src/main/java/com/lunchtime/controllers/PersonController.java:41]: Your task is - when user tries to register account with password that equals phone number or email, he sees warning

**Review by vitaliimak — CHANGES_REQUESTED** [review]

File: server/src/main/java/com/lunchtime/controllers/PersonController.java (hunk 4 above)
Comment (narekacy) [reply server/src/main/java/com/lunchtime/controllers/PersonController.java:34]: what

File: server/src/main/java/com/lunchtime/controllers/PersonController.java (hunk 5 above)
Comment (narekacy) [reply server/src/main/java/com/lunchtime/controllers/PersonController.java:41]: And what is violated in Controller?

**Review by vitaliimak — APPROVED** [review]

**Review by VozNick — APPROVED** [review]

**Review by r-yevchuk — APPROVED** [review]
